common/persistence/cassandra/cassandraClusterMetadata.go

```diff
@@ -166,20 +142,43 @@ func (m *cassandraClusterMetadata) convertPreviousMapToInitializeResponse(previo
 	}, nil
 }
 
-func (m *cassandraClusterMetadata) GetImmutableClusterMetadata() (*p.InternalGetImmutableClusterMetadataResponse, error) {
-	query := m.session.Query(templateGetImmutableClusterMetadata, constMetadataPartition)
-	var immutableMetadata []byte
+func (m *cassandraClusterMetadata) GetClusterMetadata() (*p.InternalGetClusterMetadataResponse, error) {
+	query := m.session.Query(templateGetClusterMetadata, constMetadataPartition)
+	var clusterMetadata []byte
 	var encoding string
-	err := query.Scan(&immutableMetadata, &encoding)
+	var version int64
+	err := query.Scan(&clusterMetadata, &encoding, &version)
 	if err != nil {
-		return nil, convertCommonErrors("GetImmutableClusterMetadata", err)
+		return nil, convertCommonErrors("GetClusterMetadata", err)
 	}
 
-	return &p.InternalGetImmutableClusterMetadataResponse{
-		ImmutableClusterMetadata: p.NewDataBlob(immutableMetadata, encoding),
+	return &p.InternalGetClusterMetadataResponse{
+		ClusterMetadata: p.NewDataBlob(clusterMetadata, encoding),
+		Version:         version,
 	}, nil
 }
 
+func (m *cassandraClusterMetadata) SaveClusterMetadata(request *p.InternalSaveClusterMetadataRequest) (bool, error) {
+	if request.Version == 0 {
+		query := m.session.Query(templateCreateClusterMetadata, constMembershipPartition, request.ClusterMetadata.Data, request.ClusterMetadata.Encoding.String(), 1)
+		err := query.Exec()
```

Comment:
i guess here should still use `MapScanCAS` since the query is insert if not exists

Reply:
Yes, you are right, let me change it.